### PR TITLE
[WIP]Configure Ubuntu 20.04 GCE image with GPU support

### PR DESCRIPTION
**Pull request by SubhasmitaSw**
What this PR does / why we need it:

Providing support for installing GPU drivers by default in the Ubuntu 20.04 OS image

**Additional context**
Please excuse the changes W.R.T. #907 , will be changed if merge conflict arises. 

cc @richardcase @cpanato @dims

### CODE REVIEW

File: images/capi/packer/files/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Copyright 2021 Google Inc. All rights reserved.
```
Comment (richardcase) [inline images/capi/packer/files/Dockerfile:1]: Do we need this Dockerfile?

File: images/capi/packer/files/Makefile [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2021 Google Inc. All rights reserved.
```
Comment (richardcase) [inline images/capi/packer/files/Makefile:1]: I don't think we need this either.

File: images/capi/packer/files/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
```
Comment (richardcase) [inline images/capi/packer/files/entrypoint.sh:1]: As we are using `install_gpu_driver.py` we won't need this file.

File: images/capi/packer/gce/packer.json [hunk 4]
```diff
@@ -18,7 +18,11 @@
       "source_image_family": "{{ user `source_image_family` }}",
       "ssh_username": "ubuntu",
       "type": "googlecompute",
-      "zone": "{{ user `zone` }}"
+      "zone": "{{ user `zone` }}",
+      "accelerator_type": "projects/{{user `project_id`}}/zones/{{user `zone`}}/acceleratorTypes/{{user `accelerator_type`}}",
+      "accelerator_count": "{{user `accelerator_count`}}",
```
Comment (richardcase) [inline images/capi/packer/gce/packer.json:23]: We can probably hardcode `accelerator_count` to `1` instead of making this configurable as we only need 1 gpu for the actual image build.

File: images/capi/packer/config/ansible-args.json [hunk 5]
```diff
@@ -1,3 +1,5 @@
...
+  "ansible_scp_extra_args": "\"-O\"",
+  "ansible_ssh_args": "-o IdentitiesOnly=yes -o HostKeyAlgorithms=+ssh-rsa -o PubkeyAcceptedAlgorithms=+ssh-rsa"
```
Comment (cpanato) [inline images/capi/packer/config/ansible-args.json:4]: those changes are needed? and looks like the `ansible_common_vars` changed a lot, that will not break the other images?

File: images/capi/packer/files/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    configure_nvidia_installation_dirs
+    download_nvidia_installer
+    run_nvidia_installer
+    update_cached_version
+    verify_nvidia_installation
+  fi
+  update_host_ld_cache
+}
+
+main "$@"
```
Comment (cpanato) [inline images/capi/packer/files/entrypoint.sh:180]: nit: new line

File: images/capi/packer/files/flatcar/scripts/bootstrap-flatcar.sh [hunk 7]
```diff
@@ -38,4 +38,4 @@ ln -s builder-env/bin/pip ${BINDIR}/pip
 # need to have symlink pip3 required by ansible/roles/providers/tasks/aws.yml
 ln -s builder-env/bin/pip ${BINDIR}/pip3
 
-touch ${BINDIR}/.bootstrapped
+touch ${BINDIR}/.bootstrapped
```
Comment (cpanato) [inline images/capi/packer/files/flatcar/scripts/bootstrap-flatcar.sh:41]: nit: new line

but looks like we dont need to touch on this file

File: images/capi/packer/gce/ubuntu-2004-gpu.json [hunk 8]
```diff
@@ -0,0 +1,8 @@
+{
+    "build_name": "ubuntu-2004-gpu",
+    "distribution": "Ubuntu",
+    "distribution_release": "focal",
+    "distribution_version": "2004",
+    "zone": "us-central1-a",
+    "base_image": "ubuntu-2004"
+  }
```
Comment (cpanato) [inline images/capi/packer/gce/ubuntu-2004-gpu.json:8]: nit: new line

**Comment by cpanato** [comment]
cc @codenrhoden for visibility

File: images/capi/packer/config/ansible-args.json (hunk 5 above)
Comment (SubhasmitaSw) [reply images/capi/packer/config/ansible-args.json:4]: seems we don't need them w.r.t the current standings, the image is being build perfectly fine.

File: images/capi/packer/files/Dockerfile (hunk 1 above)
Comment (SubhasmitaSw) [reply images/capi/packer/files/Dockerfile]: I've removed all unnecessary files.

File: images/capi/packer/files/flatcar/scripts/bootstrap-flatcar.sh [hunk 9]
```diff
@@ -36,6 +36,4 @@ chown -R core ${BUILDER_ENV}
 
 ln -s builder-env/bin/pip ${BINDIR}/pip
 # need to have symlink pip3 required by ansible/roles/providers/tasks/aws.yml
-ln -s builder-env/bin/pip ${BINDIR}/pip3
-
-touch ${BINDIR}/.bootstrapped
+ln -s builder-env/bin/pip ${BINDIR}/pip3
```
Comment (richardcase) [inline images/capi/packer/files/flatcar/scripts/bootstrap-flatcar.sh:39]: Does this file need to be changed? Guessing we can revert the changes?

File: images/capi/packer/config/ansible-args.json [hunk 10]
```diff
@@ -1,3 +1,3 @@
...
-}
+}
```
Comment (richardcase) [inline images/capi/packer/config/ansible-args.json:3]: Does this file need to be changed? Guessing we can revert the change or add a newline?

File: images/capi/packer/gce/ubuntu-2004-gpu.json (hunk 8 above)
Comment (richardcase) [inline images/capi/packer/gce/ubuntu-2004-gpu.json:8]: nit: should have a newline

File: images/capi/packer/gce/packer.json [hunk 11]
```diff
@@ -18,7 +18,11 @@
       "source_image_family": "{{ user `source_image_family` }}",
       "ssh_username": "ubuntu",
       "type": "googlecompute",
-      "zone": "{{ user `zone` }}"
+      "zone": "{{ user `zone` }}",
+      "accelerator_type": "projects/{{user `project_id`}}/zones/{{user `zone`}}/acceleratorTypes/{{user `accelerator_type`}}",
```
Comment (aniruddha2000) [inline images/capi/packer/gce/packer.json:22]:
```suggestion
      "accelerator_type": "projects/{{ user `project_id` }}/zones/{{ user `zone` }}/acceleratorTypes/{{ user `accelerator_name` }}",
```

File: images/capi/packer/gce/packer.json [hunk 12]
```diff
@@ -83,6 +92,7 @@
   "variables": {
     "ansible_common_vars": "",
     "ansible_extra_vars": "",
+    "ansible_scp_extra_args": "",
```
Comment (aniruddha2000) [inline images/capi/packer/gce/packer.json:95]:
```suggestion
    "ansible_scp_extra_args": "",
    "accelerator_name": "{{ user `accelerator_name` }}",
```

File: images/capi/packer/gce/packer.json (hunk 11 above)
Comment (aniruddha2000) [reply images/capi/packer/gce/packer.json:22]: `accelerator_type` will get conflict.

**Comment by SubhasmitaSw** [comment]
Since it was merged with the [18.04 version pr](https://github.com/kubernetes-sigs/image-builder/pull/934), I'm closing this.
